src/NzbDrone.Core/Indexers/Rarbg/RarbgParser.cs

```diff
@@ -17,6 +18,10 @@ public IList<ReleaseInfo> ParseResponse(IndexerResponse indexerResponse)
 
             switch (indexerResponse.HttpResponse.StatusCode)
             {
+                case HttpStatusCode.TooManyRequests:
+                    throw new RequestLimitReachedException("Indexer API limit reached", TimeSpan.FromMinutes(2));
+                case (HttpStatusCode)520:
+                    throw new RequestLimitReachedException("Indexer API returned unknown error", TimeSpan.FromMinutes(3));
```

Comment:
Nit: 520 is known - internal servers behind cloudflare are rate limited/overwhelmed

```
when rated limited web returns http code 444 to cloudflare which returns 520 on cloudflare , there is no way to send 429 to cloudflare without it breaking the request properly and retrying it
```

Reply:
Unless it throws a 520 for another reason…